test/jdk/java/util/Collection/MOAT.java

```diff
@@ -564,6 +565,12 @@ private static void testListMutatorsAlwaysThrow(List<Integer> c) {
                 () -> c.addAll(0, Collections.emptyList()));
     }
 
+    private static void testImmutableListMutatorsAlwaysThrow(List<Integer> c) {
+        THROWS(UnsupportedOperationException.class,
+                c::removeFirst,
+                c::removeLast);
+    }
+
```

Comment:
Would this test better if inserted in `testImmutableCollection(final Collection<T> c, T t)`, Line 477'ish.
Or in `testImmutableList(final List<Integer> c)`, line 519.

Reply:
Unfortunately, the `remove(F|L)` methods are not available for general collections. `testImmutableList()` is also applied for unmodifiable lists (like Collections.emptyList() and Collections.nCopies(0, ...)) that seem to behave differently (they throw NSEE). The latter have different specifications compared to an immutable list.